drizzlepac/haputils/catalog_utils.py

```diff
@@ -2174,6 +2188,17 @@ def measure_sources(self, filter_name):
             self.source_cat = SourceCatalog(imgarr_bkgsub, self.sources, background=self.image.bkg_background_ra,
                                                 error=total_error, kernel=self.kernel, wcs=self.image.imgwcs)
 
+        # Guard against a non-standard coordinate system being defined in the input image headers
+        # If something non-standard is found, set the value to 'ICRS' to insure that the
+        # table conversion works.  Checking against the set of valid options recognized by WCSLIB and
+        # documented in http://ds9.si.edu/doc/ref/region.html#RegionFileFormat is necessary since the
+        # header keyword REFFRAME can be populated with anything specified by the user in the original
+        # proposal.
+        if self.source_cat._wcs.wcs.radesys.upper() not in RADESYS_OPTIONS:
+            self.source_cat._wcs.wcs.radesys = 'ICRS'
+            log.warning(f"Assuming input coordinates are ICRS, instead of {self.source_cat._wcs.wcs.radesys}")
+            log.warning(f"Sky coordinates of source objects may not be accurate.")
+
```

Comment:
Solution looks fine, but since it has to be done twice...is it easy to make a little helper method do to this?

Reply:
Implemented under a71cb83.